core/data_visualization.py

```diff
@@ -53,25 +80,46 @@ def plot_hist(x, bins, x_axis, y_axis, file_name):
         file_name,
     )
 
-def generate_basic_plots(features, label):
+def generate_scatter_plots(features, label):
     for column in features.columns:
         x = features.loc[:, column]
-        num_column_unique_values = features[column].nunique()
-        if num_column_unique_values < MIN_REAL_FEATURE_UNIQUE_VALUES:
-            plot_hist(x, bins=num_column_unique_values, x_axis=column, y_axis=label.name, file_name=f'hist_{column}')
-        else:
-            plot_scatter(x, label, x_axis=column, y_axis=label.name, file_name=f'scatter_{column}')
+        plot_scatter(x, label, x_axis=column, y_axis=label.name, file_name=f'scatter_{column}')
+
+def generate_histogram_plots(features):
+    for column in features.columns:
+        x = features.loc[:, column]
+        plot_hist(x, bins=features[column].nunique(), x_axis=column, y_axis='count', file_name=f'hist_{column}')
 
 def generate_compound_plots(features, label):
     for column1, column2 in combinations(features.columns, r=2):
-        x1 = features.loc[:, column1]
-        x2 = features.loc[:, column2]
-        x = x1 / x2
+        x = features.loc[:, column1]
+        y = features.loc[:, column2]
+
+        x_axis = f'{column1}-{column2}'
+        plot_scatter(x / y, label, x_axis=x_axis, y_axis=label.name, file_name=f'compound_{x_axis}')
 
-        compound_column = f'{column1}-{column2}'
-        plot_scatter(x, label, x_axis=compound_column, y_axis=label.name, file_name=f'scatter_{compound_column}')
-        plot_scatter(x1, x2, x_axis=column1, y_axis=column2, file_name=f'correlation_{compound_column}')
+def generate_correlation_plots(features):
+    for column1, column2 in combinations(features.columns, r=2):
+        x = features.loc[:, column1]
+        y = features.loc[:, column2]
+
+        plot_scatter(x, y, x_axis=column1, y_axis=column2, file_name=f'correlation_{column1}-{column2}')
+
+def generate_classification_plots(features, label):
+    for column1, column2 in combinations(features.columns, r=2):
+        x = features.loc[:, column1]
+        y = features.loc[:, column2]
 
-def generate_data_visualization_plots(features, label):
-    generate_basic_plots(features, label)
-    generate_compound_plots(features, label)
+        classes = _get_classes(x, y, label)
+        plot_classification(
+            classes.keys(),
+            [
+                [x for x, _ in c] for c in list(classes.values())
+            ],
+            [
+                [y for _, y in c] for c in list(classes.values())
+            ],
```

Comment:
black magic version
```suggestion
            *zip(*classes.values()),
```

Reply:
Doesn't seem to work when I test it.

Reply:
God I misread it's cause it's supposed to be even worse than I thought
```py
*zip(*(zip(*points) for points in classes.values())),
```

Reply:
Oh god, I love it!